### DISCUSSION TITLE
How to insert a Texture as an Asset?

### DISCUSSION THREAD

**Discussion by hangvane** [bevyengine/bevy#11547] [Q&A]
Hi, I'm trying to build a custom rendering pipeline to an offscreen texture for later modification. I followed the guidence of GPT4 but I can't insert a ``Texture`` as an ``Asset``:
```rust
fn setup_texture(
    mut commands: Commands,
    render_device: Res<RenderDevice>,
    mut textures: ResMut<Assets<Texture>>,
) {
    let offscreen_texture = render_device.create_texture(....);
    let texture_handle: Handle<Texture> = textures.add(offscreen_texture);
}
```
The error is:
```shell
86  |     mut textures: ResMut<Assets<Texture>>,
    |                          ^^^^^^^^^^^^^^^ the trait `Asset` is not implemented for `bevy::render::render_resource::Texture`
```
Is it possible to insert a ``Texture`` as an ``Asset``?

If not, what is the best practice to save multiple instances of ``Texture``?

**Comment by nicopap** [comment]
Beware that GPT4 is trained on out-of-date versions of bevy, it is unreliable when it comes to the bevy API.

 It is not possible to insert a `Texture` as an asset.

You say you want:

> a custom rendering pipeline to an offscreen texture for later modification

Have you considered using [Render to texture](https://bevyengine.org/news/bevy-0-7/#render-to-texture)? It allows using a camera view as a texture, if what is in the camera field of view changes, then the texture changes.

Alternatively, consider using the [screenshot API](https://bevyengine.org/news/bevy-0-11/#screenshot-api) if you want to re-use a static image. [`take_screenshot`](https://docs.rs/bevy/latest/bevy/render/view/window/screenshot/struct.ScreenshotManager.html#method.take_screenshot) seems to be the method you would need to use. An `Image` is an asset you can add to `Assets<Image>`.
